csrc/scheduler/mma_utils.h

```diff
@@ -170,12 +170,44 @@ class WarpMmaSwizzler {
       TensorView* tv,
       MmaOptions options = MmaOptions());
 
+  //! Note [schedule of ldmatrix]
+  //! If you look at the doc of ldmatrix and mma for Turing and Ampere:
+  //! https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#warp-level-matrix-fragment-mma-16816-float
+  //! https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#warp-level-matrix-instructions-ldmatrix
+  //! you will find that, the memory layout of the output of ldmatrix, which
+  //! matches with the input layout of MMA instruction, mismatch with the index
+  //! that each thread uses to call ldmatrix. In nvFuser, we schedule the
+  //! allocation domain of the ldmatrix output and mma inputs to be consistent
+  //! with the memory layout of the output of ldmatrix, and we schedule the
+  //! leaf domain of the ldmatrix output to be consistent with the index that
+  //! each thread uses to call ldmatrix. This function is used to schedule the
+  //! leaf domain of the ldmatrix output. The allocation domain of the ldmatrix
+  //! output and mma inputs are scheduled in scheduleOperandRead, which must be
+  //! called before this function.
+  //!
+  //! ldmatrix loads multiple 8x8 matrices from shared memory to registers in a
+  //! swizzled memory format.
+  //!   +--------+--------+
+  //!   |        |        |
+  //!   |  8x8   |  8x8   |
+  //!   |        |        |
+  //!   +--------+--------+
+  //!   |        |        |
+  //!   |  8x8   |  8x8   |
+  //!   |        |        |
+  //!   +--------+--------+
+  //! If mn_major is true, these 8x8 matrices are visited in the order of:
+  //! top left -> top right -> bottom left -> bottom right.
+  //! If mn_major is false, these 8x8 matrices are visited in the order of:
+  //! top left -> bottom left -> top right -> bottom right.
+  static void scheduleLdMatrix(TensorView* tv, bool mn_major = false);
```

Comment:
Based on the changes in `matmul.cpp`, the `mn_major` swizzle pattern is applied for loading `B` input of `MmaOp`.

Can you briefly share the background for this, or it will be enough to revisit the links you shared in the documentation?

Reply:
This is a very good question. I updated the code to add a paragraph of comments explaining the situation. Cross posting it here:
```C++
  //! In principle, only `mn_major = false` should be needed. But unfortunately,
  //! we are taking advantage of the ldmatrix large load in a pretty hacky way.
  //! For example, for Turing, only m16n8k8 is supported by hardware. But we are
  //! also using a fake m16n8k16 and m16n16k16, which uses a single large
  //! ldmatrix to load data to register, and run multiple mma instructions to
  //! consume these data. In the future, we should only keep the m16n8k8 macro,
  //! and schedule m16n8k16 and m16n16k16 more correctly than this current way.
```

Reply:
Thank you for adding this piece of comment, now everything makes sense!